sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -433,28 +610,45 @@ double SCIStrategicPlugin::findSpeedLimit(const lanelet::ConstLanelet& llt) cons
   }
 }
 
-void SCIStrategicPlugin::generateMobilityOperation()
+cav_msgs::MobilityOperation SCIStrategicPlugin::generateMobilityOperation()
 {
     cav_msgs::MobilityOperation mo_;
-    mo_.header.timestamp = ros::Time::now().toNSec() * 1000000;
-
-    std::string id(bsm_id.begin(), bsm_id.end());
-    mo_.header.sender_bsm_id = id;
+    mo_.header.timestamp = ros::Time::now().toNSec()/1000000;
+    mo_.header.sender_id = config_.vehicle_id;
+    mo_.header.sender_bsm_id = bsm_id_;
+    mo_.strategy = stop_controlled_intersection_strategy_;
 
-    int flag = (approaching_stop_controlled_interction_ ? 1 : 0);
+    int flag = (is_allowed_int_ ? 1 : 0);
```

Comment:
Should just be able to assign a bool value to an int for an implicit conversion, and vice-versa.

Reply:
Updated.